packages/data-lake-api/src/DataLakeEventsFetchQuery.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { SqlQuery } from './SqlQuery';
+
+export class DataLakeEventsFetchQuery {
+  constructor(database, options) {
+    this.database = database;
+
+    const {
+      organisationUnitCodes,
+      startDate,
+      endDate,
+      eventId,
+      dataGroupCode,
+      dataElementCodes,
+    } = options;
+
+    this.dataElementCodes = dataElementCodes;
+    this.entityCodes = organisationUnitCodes;
+    this.startDate = startDate;
+    this.endDate = endDate;
+    this.eventId = eventId;
+    this.dataGroupCode = dataGroupCode;
+    this.hasDataElements = dataElementCodes && dataElementCodes.length > 0;
+  }
+
+  async fetch() {
+    const { query, params } = this.buildQueryAndParams();
+
+    const sqlQuery = new SqlQuery(query, params);
+
+    return sqlQuery.executeOnDatabase(this.database);
+  }
+
+  getAliasedColumns() {
+    const aliasedColumns = [
+      'date',
+      'entity_code AS "entityCode"',
+      'entity_code AS "entityName"',
```

Comment:
Just using `entity_code` as `name` here. Personally I'm a bit torn on how we want to handle this in the future.

We could fairly easily query the database to get the entity names from the code, however that alone really highlights to me how redundant the `entityName` field really is... Going forward I feel I'd prefer we omitted it entirely... thoughts?

Reply:
Agree, lets force the issue by returning empty string instead?

Reply:
Oooh good idea! Will do 👍